graphql/achievement/mutations.ts

```diff
@@ -0,0 +1,33 @@
+import { Arg, Authorized, Ctx, Mutation, Resolver } from 'type-graphql';
+import { User_achievement as Achievement } from '../generated';
+import { Role } from '../roles';
+import * as Notification from '../../common/notification';
+import { GraphQLContext } from '../context';
+import { AuthorizedDeferred, hasAccess } from '../authorizations';
+import { prisma } from '../../common/prisma';
+
+@Resolver(() => Achievement)
+export class MutateAchievementResolver {
+    @Mutation((returns) => Boolean)
+    @AuthorizedDeferred(Role.ADMIN, Role.OWNER)
+    async matchMeetingJoin(@Ctx() context: GraphQLContext, @Arg('matchId') matchId: number) {
+        const { user } = context;
+        const match = await prisma.match.findUnique({ where: { id: matchId }, include: { pupil: true, student: true } });
+        await hasAccess(context, 'Match', match);
+
+        if (user.studentId) {
+            await Notification.actionTaken(user, 'student_joined_match_meeting', {
+                matchId: matchId.toString(),
+                pupil: match.pupil,
+                relation: `match/${matchId}`,
+            });
+        } else if (user.pupilId) {
+            await Notification.actionTaken(user, 'pupil_joined_match_meeting', {
+                matchId: matchId.toString(),
+                relation: `match/${matchId}`,
```

Comment:
Could you please create a helper function next to this function, where the split is happening

https://github.com/corona-school/backend/pull/866/files#diff-98b7fdaff6dad941d9abe0a45b8b1e2a43ff4bacc1acd99bd3c5a1479356a325R26-R33

And replace it everywhere